### PR TITLE
feat: login/register using GitHub

### LINKED ISSUE #125: feat: signin/register using github account

### PR DESCRIPTION
**Pull request by ShivamMenda**
Closes #125

### CODE REVIEW

**Comment by ShivamMenda** [comment]
https://github.com/AOSSIE-Org/Resonate/assets/74780977/d9bc80d3-2c14-4fc4-88bd-c785bcb69dd4

**Comment by ShivamMenda** [comment]
@chandansgowda Implemented the GitHub login. For now, I generated an Oauth app on my personal GitHub account, We can replace it later.

File: lib/views/screens/signup_screen.dart [hunk 1]
```diff
@@ -220,6 +220,35 @@ class _SignupScreenState extends State<SignupScreen> {
                     ],
                   ),
                 ),
+                SizedBox(height: UiSizes.height_10),
+                ElevatedButton(
```
Comment (chandansgowda) [inline lib/views/screens/signup_screen.dart:224]: Please refactor it to a different file under widgets. We are using the same code four times. 2 in login and 2 in signup screen.

**Review by chandansgowda — CHANGES_REQUESTED** [review]

File: lib/views/screens/signup_screen.dart (hunk 1 above)
Comment (ShivamMenda) [reply lib/views/screens/signup_screen.dart:224]: Sure @chandansgowda will resolve it.

Comment (ShivamMenda) [reply lib/views/screens/signup_screen.dart]: @chandansgowda Resolved.

**Comment by ShivamMenda** [comment]
@chandansgowda Refactoring is done. We can now put up the App Id and app client from the aossie account.

**Comment by chandansgowda** [comment]
@ShivamMenda Reduce the size of AOSSIE logo a bit on both screens. Coz the bottom half of the screen looks a bit clumsy.

**Review by chandansgowda — CHANGES_REQUESTED** [review]
@ShivamMenda Reduce the size of AOSSIE logo a bit on both screens. Coz the bottom half of the screen looks a bit clumsy.

**Comment by ShivamMenda** [comment]
@chandansgowda Reduced the size.

**Comment by chandansgowda** [comment]
@ShivamMenda Looks good. Check my comments on #132. I'll merge this PR once that is merged.

**Comment by ShivamMenda** [comment]
> @ShivamMenda Looks good. Check my comments on #132. I'll merge this PR once that is merged.

Sure. Will resolve that. We need to change the app client id and secret on the appwrite console once merged.

**Comment by chandansgowda** [comment]
I have changed the secrets on appwrite.

**Comment by chandansgowda** [comment]
Great job @ShivamMenda 🚀

**Pull request merged**